demo/js/src/main/scala/games/audio/Context.scala

```diff
@@ -5,53 +5,71 @@ import org.scalajs.dom
 import games.Resource
 import games.math.Vector3f
 
+import java.nio.ByteBuffer
+
+import scala.collection.mutable.Set
+
 import js.Dynamic.{ global => g }
 
-class JsContext extends Context {
-  val audioContext: js.Dynamic = {
-    val classicAudioContext: js.UndefOr[js.Dynamic] = g.AudioContext
-    classicAudioContext.toOption match {
-      case Some(x) => x
-      case None => {
-        val webKitAudioContext: js.UndefOr[js.Dynamic] = g.webkitAudioContext
-        webKitAudioContext.toOption match {
-          case Some(x) => x
-          case None    => throw new RuntimeException("Web Audio API not supported by your browser")
-        }
-      }
-    }
-  }
+class WebAudioContext extends Context {
+
+  val classicAudioContext: js.UndefOr[js.Dynamic] = g.AudioContext
+  val webKitAudioContext: js.UndefOr[js.Dynamic] = g.webkitAudioContext
+  val audioContext: js.Dynamic = classicAudioContext.orElse(webKitAudioContext).getOrElse(throw new RuntimeException("Web Audio API not supported by your browser"))
```

Comment:
This will create three fields in the class. You should write it like this:
```scala
val audioContext: js.Dynamic = {
  val classicAudioContext: js.UndefOr[js.Dynamic] = g.AudioContext
  val webKitAudioContext: js.UndefOr[js.Dynamic] = g.webkitAudioContext
  classicAudioContext.orElse(webKitAudioContext).getOrElse(
      throw new RuntimeException("Web Audio API not supported by your browser"))
}
```